controller/model/authenticator_mod_cert.go

```diff
@@ -81,13 +89,55 @@ func (module *AuthModuleCert) Process(context AuthContext) (string, error) {
 				}
 			}
 
-			return authenticator.IdentityId, nil
+			opts := x509.VerifyOptions{
+				Roots:         module.getRootPool(),
+				Intermediates: x509.NewCertPool(),
+				KeyUsages:     []x509.ExtKeyUsage{x509.ExtKeyUsageAny},
+			}
+
+			if _, err := cert.Verify(opts); err == nil {
+				return authenticator.IdentityId, nil
+			} else {
+				println(err)
+			}
 		}
 	}
 
 	return "", apierror.NewInvalidAuth()
 }
 
+func (module *AuthModuleCert) getRootPool() *x509.CertPool {
+	roots := x509.NewCertPool()
+
+	for _, caCert := range module.staticCaCerts {
+		roots.AddCert(caCert)
+	}
+
+	cas, err := module.env.GetHandlers().Ca.Query("isAuthEnabled = true and isVerified = true limit 500")
```

Comment:
i can't imagine a world where there's 500 ca's in one instance of ziti but... dumb things happen maybe? is there any harm in going moar bigger? why stop at 500 when it seems (to me from the code) that this is "just give me all of them"

Reply:
data net enrolls every customer's CA as 3rd party ca and auto-enrolls the probes

Reply:
Should be a stream of CAs, fixing.